kotlin/viewregistry-android/src/main/java/com/squareup/viewregistry/PanelContainer.kt

```diff
@@ -0,0 +1,92 @@
+package com.squareup.viewregistry
+
+import android.R.style
+import android.app.Dialog
+import android.content.Context
+import android.graphics.Color
+import android.graphics.drawable.ColorDrawable
+import android.support.annotation.ColorInt
+import android.util.AttributeSet
+import android.view.ViewGroup
+import android.view.ViewGroup.LayoutParams.MATCH_PARENT
+import android.view.WindowManager.LayoutParams.FLAG_DIM_BEHIND
+import android.view.WindowManager.LayoutParams.FLAG_FULLSCREEN
+import android.view.WindowManager.LayoutParams.SOFT_INPUT_ADJUST_PAN
+import com.squareup.viewregistry.R.layout
+import io.reactivex.Observable
+import kotlin.math.min
+import kotlin.reflect.jvm.jvmName
+
+/**
+ * Container view for [PanelContainerScreen].
+ */
+class PanelContainer
+@JvmOverloads constructor(
+  context: Context,
+  attributeSet: AttributeSet? = null
+) : AbstractModalContainer<BackStackScreen<*>>(context, attributeSet) {
+
+  /**
+   * Always returns `true`, because a view created to display one [BackStackScreen]
+   * can display any [BackStackScreen].
+   */
+  override fun BackStackScreen<*>.matches(modalScreen: BackStackScreen<*>) = true
+
+  override fun showDialog(
+    modalScreen: BackStackScreen<*>,
+    screens: Observable<out BackStackScreen<*>>,
+    viewRegistry: ViewRegistry
+  ): Dialog {
+    val binding = viewRegistry.getBinding<BackStackScreen<*>>(modalScreen::class.jvmName)
+    val view = binding.buildView(screens, viewRegistry, this)
+
+    // TODO Could we instead define a theme based on, say, style.Theme_Light,
+    // to define *everything*, including entry and exit animation? e.g., this
+    // was working nicely except that I couldn't make the window background transparent.
+    // Didn't try hard, though. Probably would still have to use code to force body
+    // width / height, though, since that needs to be based on screen geometry, regardless
+    // of what shape the window is forced into by the soft keyboard.
+    /*
+    return Dialog(context, style.Theme_Light).apply {
+      setContentView(layout.panel_container)
+      findViewById<ViewGroup>(R.id.panelBody).addView(view)
+      show()
+      window.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
+    }
+    */
+
+    return Dialog(context).apply {
+      setContentView(layout.panel_container)
+      setCancelable(false)
```

Comment:
Unclear what a `Panel` is and how it differs from a `Modal`. Why is a panel always un-cancelable while a modal is configurable?

Reply:
I haven't finished reading through this PR yet but my main question so far is "what is a PanelContainerScreen and what's a Panel?"

Reply:
Ok, just got to the bottom, and I still have no idea.

Reply:
Somewhere along the way I started thinking about PanelContainer and friends as sample code rather than library code, but I left it here in the wrong place. Basically, it became an _example_ of how one can write a container for their preferred style of custom modals. If I stick with that approach, it should move to the sample app and be documented as "A screen in a modal subflow, shown full screen on phones and in a square overlay on tablets." Basically, I was trying to reproduce how subflows look in Square PoS.

But before merging I'll take a step back and see if it can't be made more generic — maybe we can let the layout params of the custom view dictate the size of the dialog? 

If that's practical, I'll change the Panel* names to something like CustomModal*. If I can't find that fairly quickly, I'll move Panel* out to sample land, and file a follow up ticket to at least try harder to find it. But really, if tiny custom modal container subclasses is what apps have to do, I could live with that.